kong/runloop/handler.lua

```diff
@@ -574,6 +576,12 @@ return {
   rewrite = {
     before = function(ctx)
       ctx.KONG_REWRITE_START = get_now()
+
+      -- special handling for proxy-authorization and te headers in case
+      -- the plugin(s) want to specify them (store the original)
+      ctx.http_proxy_authorization = ngx.var.http_proxy_authorization
```

Comment:
We should be able to cache `ngx.var` at module level variable since they are frequently used. They work through the magic of metamethods and can be safely cached.

Reply:
Yes, `ngx.var` is not a problem, but `ngx.var.value` is. With `ngx.ctx` it is the otherway around. So I think caching that at the module level doesn't gain much as `ngx` is already cached (like @p0pr0ck5 mentions below).

Reply:
Not sure what you mean exactly, but the `ngx.var` table itself does not contain the actual value of the variables, they are retrieved dynamically using the `__index` metamethod. So the table is always empty and caching it will make all access to it having one less indirections, and should be faster than just caching `ngx` alone.

Reply:
Local caching of globals is what is usually recommended, but caching local table (in this case localized ngx) functions is what I don't think adds anything (it could add tinyt bits, but very very tiny that goes to error margin of measuremebt). This is explained here: http://luajit.org/ext_ffi_tutorial.html#cache

Reply:
@dndx / @p0pr0ck5,

I made another commit to localize all the things in `runloop/handler.lua`:
https://github.com/Kong/kong/pull/4316/commits/4502523ce96b4881f4025f83b06abb14472545bc

Please let me know if we want or not want this. Also @thibaultcha may have something to say about it.

Reply:
Crap, our unit test suite is not happy with localizing funcs:
https://travis-ci.org/Kong/kong/builds/521608375

Reply:
Oh, actually a typo at localizing one function. Fixed now. What do you think? With or without?